src/dodal/devices/fast_grid_scan.py

```diff
@@ -243,6 +229,15 @@ def __init__(self, prefix: str, smargon_prefix: str, name: str = "") -> None:
         }
         super().__init__(name)
 
+    async def _calculate_expected_images(self):
+        x = await self.x_steps.get_value()
+        y = await self.y_steps.get_value()
+        z = await self.z_steps.get_value()
```

Comment:
Could: Can this be parallelized? If not then may be worth putting a comment explaining why to avoid tripping someone up in future.

```suggestion
        x, y, z = await asyncio.gather(self.x_steps.get_value(), self.y_steps.get_value(), self.z_steps.get_value()
```

Reply:
Yes, it can be parallelised. I have tended not to bother with parallel reads because they're normally very short anyway but I will add it in.

Reply:
Fair enough, my concern is mainly that someone in the future might wonder why it wasn't done and go down an archaeological rabbit hole.